lib/miq_expression/target.rb

```diff
@@ -48,14 +48,18 @@ def plural?
     [:has_many, :has_and_belongs_to_many].include?(reflections.last.macro)
   end
 
+  def virtual_reflection?
+    reflections != collect_reflections
+  rescue ArgumentError
```

Comment:
Whose error-prone arguments are we anticipating here? This seems oddly placed, as there are no arguments being passed. Perhaps it belongs further down?